port/cpl_aws.cpp

```diff
@@ -1093,38 +1094,62 @@ static bool ReadAWSCredentials(const std::string &osProfile,
     osAccessKeyId.clear();
     osSessionToken.clear();
 
-    VSILFILE *fp = VSIFOpenL(osCredentials.c_str(), "rb");
-    if (fp != nullptr)
+    static std::mutex goMutex;
+    std::lock_guard oGuard(goMutex);
+
+    // Cache last request and result to avoid repeated file accesses.
```

Comment:
Is the credentials cache from`osCredentialsFilename`cache populated once at process start and never again? Users could (should) update the value periodically, so a long-running GDAL processes might get access denied from AWS but it's possible new credentials were updated in the file.

Reply:
> Is the credentials cache from osCredentialsFilename cache populated once at process start and never again?

yes

> but it's possible new credentials were updated in the file.

is it really realistic that ~/.aws/config or ~/.aws/credentials content change during the lifetime of a process ? Aren't those used for 'static' credentials ?

Reply:
> is it really realistic that ~/.aws/config or ~/.aws/credentials content change during the lifetime of a process ? Aren't those used for 'static' credentials ?

The typical GDAL user would probably not rotate credentials in the lifetime of a process (such as in a single gdalwarp call).  However, some companies/AWS IAM policies require users to rotate static credentials every few hours.  My particular use case (long-running Mapserver) is probably an edge case where it's easy enough for me to restart a Mapserver FastCGI process.

Reply:
ok, so this basically means that the whole idea is busted. I'll revert that commit and just keep the CPLGetPhysicalRAM() change